portals/publisher/src/main/webapp/source/src/app/components/Shared/PoliciesUI/PolicyDropzone.tsx

```diff
@@ -93,76 +93,134 @@ interface PolicyDropzoneSharedProps {
     PolicyConfiguringDrawer: any;
 }
 
-const PolicyDropzoneShared: FC<PolicyDropzoneSharedProps> = ({
-    policyDisplayStartDirection,
-    currentPolicyList,
-    setCurrentPolicyList,
-    currentFlow,
-    target,
-    verb,
-    allPolicies,
-    isAPILevelPolicy,
-    drop,
-    canDrop,
-    droppedPolicy,
-    setDroppedPolicy,
-    AttachedPolicyList,
-    PolicyConfiguringDrawer
-}) => {
+// Option 1: `listOriginatedFromCommonPolicies` and `isApiRevision` are provided
+interface PolicyDropzoneWithCommonProps extends PolicyDropzoneSharedBaseProps {
+    listOriginatedFromCommonPolicies: string[];
+    isApiRevision: boolean;
+}
+
+// Option 2: Neither `listOriginatedFromCommonPolicies` nor `isApiRevision` are provided
+interface PolicyDropzoneWithoutCommonProps extends PolicyDropzoneSharedBaseProps {
+    listOriginatedFromCommonPolicies?: undefined;
+    isApiRevision?: undefined;
+}
 
-    return (
-        (<Root>
-            <Grid container>
-                <div
-                    ref={drop}
-                    className={clsx({
-                        [classes.dropzoneDiv]: true,
-                        [classes.acceptDrop]: canDrop,
-                        [classes.alignCenter]: currentPolicyList.length === 0,
-                        [classes.alignLeft]:
-                            currentPolicyList.length !== 0 &&
-                            policyDisplayStartDirection === 'left',
-                        [classes.alignRight]:
-                            currentPolicyList.length !== 0 &&
-                            policyDisplayStartDirection === 'right',
-                    })}
-                >
-                    {currentPolicyList.length === 0 ? (
-                        <Typography>
-                            <FormattedMessage
-                                id='App.Components.Policies.Drop.Zone.text.label'
-                                defaultMessage='Drag and drop policies here'
+// Combine the two using a union type
+type PolicyDropzoneSharedProps = PolicyDropzoneWithCommonProps | PolicyDropzoneWithoutCommonProps;
+
+const PolicyDropzoneShared: FC<PolicyDropzoneSharedProps> = (props) => {
```

Comment:
Destruct props

Reply:
relates to https://github.com/wso2/apim-apps/pull/809#discussion_r1901575381